airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -282,13 +282,17 @@ def _get_ti_pod_labels(context: Optional[dict] = None, include_try_number: bool
         if not context:
             return {}
 
-        labels = {
-            'dag_id': context['dag'].dag_id,
-            'task_id': context['task'].task_id,
-            'execution_date': context['ts'],
-        }
+        ti = context['ti']
+
+        labels = {'dag_id': ti.dag_id, 'task_id': ti.task_id, 'execution_date': context['ts']}
```

Comment:
probably we should replace `execution_date` with `run_id` here, since that is now the "real" PK component right?.  the signifigance of these labels, AFAIK, is just for "reattaching" to a pod after scheduler death, so it's just meant to be attrs that uniquely identify the TI.  though it needn't necessarily be done in this PR of course.

Reply:
I had that at first here but it makes the change about task mapping hard to see, so I've got it in a separate commit/pr

(If you haven't noticed I'm a big fan of Pr=one thing)

Reply:
yup, makes sense

Reply:
https://github.com/apache/airflow/pull/21960